### PR TITLE
Archiving products (SHOOP-1982)

### PR DESCRIPTION
**Pull request by shawnadelic**
Add product archiving to Core and Admin.

Refs SHOOP-1982

### CODE REVIEW

File: shoop/core/models/_products.py [hunk 1]
```diff
@@ -163,6 +164,7 @@ class Product(TaxableItem, AttributableMixin, TranslatableModel):
     created_on = models.DateTimeField(auto_now_add=True, editable=False, verbose_name=_('created on'))
     modified_on = models.DateTimeField(auto_now=True, editable=False, verbose_name=_('modified on'))
     deleted = models.BooleanField(default=False, editable=False, db_index=True, verbose_name=_('deleted'))
+    archived = models.BooleanField(default=False, editable=False, db_index=True, verbose_name=_('archived'))
```
Comment (tulimaki) [inline shoop/core/models/_products.py]: There shouldn't be need to create new attribute for archiving. I think you should use delete for archiving. Also use the soft delete functionality to archive/hide products.

**Comment by suutari-ai** [comment]
Maybe you should have just renamed the "Delete" button to "Archive" in UI?

I don't think we need two flags for excluding products from listings. Just one should be enough. The name for that field could be "deleted" in the backend as it is now (to avoid a migration). Setting that flag is effectively just archiving, since nothing is really deleted and it could even be brought back by clearing the flag.

**Comment by shawnadelic** [comment]
Sure, that would be easy enough, I had just assumed that "archive" would have a separate use case from "delete," but essentially they are doing the same thing.

**Comment by suutari-ai** [comment]
The current functionality for "soft delete" is good as it is.

**Comment by suutari-ai** [comment]
... and @kmn1 said that "Delete" is better term in the UI than "Archive", so no need to change that either.

**Pull request closed without merging**